src/tools/compiletest/src/header.rs

```diff
@@ -809,6 +811,12 @@ fn expand_variables(mut value: String, config: &Config) -> String {
         value = value.replace(BUILD_BASE, &config.build_base.to_string_lossy());
     }
 
+    if value.contains(RUST_SRC_BASE) {
+        let src = config.sysroot_base.join("lib/rustlib/src/rust");
+        let canonical = canonicalize(&src).unwrap();
+        value = value.replace(RUST_SRC_BASE, &canonical.to_string_lossy());
+    }
+
```

Comment:
This is required, rather than just having a `{{sysroot-base}}` template, because `/home/me/rust/build/<target>/stage1/lib/rustlib/src/rust` is a symlink to `/home/me/rust/`. When `/rust/$hash` is mapped back to a real path, we resolve symlink by one step: https://github.com/rust-lang/rust/blob/2f8d81f9dbac6b8df982199f69da04a4c8357227/compiler/rustc_session/src/config.rs#L2838-L2850, so it becomes `/home/me/rust/.../foo.rs`. If we were to have a `--remap-path-prefix={{sysroot-base}}=remapped` in the test header, nothing will get remapped because it becomes `--remap-path-prefix=/home/me/rust/build/<target>/stage1/=remapped`, which doesn't match `/home/me/rust/`